GLOSSARY.md

```diff
@@ -0,0 +1,128 @@
+# Glossary
+This is a glossary of terminology (possibly) used in the chalk crate.
+
+## Binary connective
+There are sixteen logical connectives on two boolean variables. The most
+interesting in this context are listed below. There is also a truth table given
+which encodes the possible results of the operations like this
+
+```
+f(false, false) f(false, true) f(true, false) f(true, true).
+```
+
+As a shorthand the resulting truth table is encoded with `true = 1` and `false =
+0`.
+
+```
+| Truth table | Operator symbol | Common name                      |
+|-------------|-----------------|----------------------------------|
+| 0001        | &&              | Conjunction; and                 |
+| 0111        | ||              | Disjunction; or                  |
+| 1001        | <=>             | Equivalence; if and only if; iff |
+| 1101        | =>              | Implication; if ... then         |
+```
+
+## Binder
+A binder is an expression that binds a variable to a certain expression.
+Examples for binders:
+
+- The universal quantifier `forall(a)` states that a certain condition holds for
+  all allowed values for `a`.
+- A function definition `f(x) = a * x` is a binder for the variable `x` whereas
+  `a` is a free variable.
+- A sum `\sum_n x_n` binds the index variable `n`.
+
+## Canonical
+## Clause
+In the A clause is the disjunction of several expressions. For example the clause
+`condition_1 || condition_2 || ...` states that at least one of the conditions
+holds.
+
+There are two notable special cases of clauses. A *Horn clause* has at most one
+positive literal. A *Definite clause* has exactly one positive literal.
+
+## Datum
+(empty)
+
+## DeBrujin Index
+(empty)
+
```

Comment:
Once you have filled this section, you may want to move the Canonical section just after this one, since in particular Canonical deals with rebounding variables to lower DeBruijn indices.